### DISCUSSION TITLE
What are the benefits and drawbacks of namespacing components?

### DISCUSSION THREAD

**Discussion by veloware** [chakra-ui/chakra-ui#8442] [General]
From looking at https://github.com/chakra-ui/chakra-ui/blob/main/MIGRATION.md#namespace-components and also digging through the v3 source code, there's a movement towards namespacing components, but I notice the components are also exposed individually - 

e.g.

```
...
export { AlertTitle } from "./alert-title"
export type { AlertTitleProps } from "./alert-title"

export * as Alert from "./namespace"
```

and then in `namespace.ts` - 

```
export { AlertTitle as Title } from "./alert-title"
export type { AlertTitleProps as TitleProps } from "./alert-title"
```

So if I understand that correctly, I can do all of the following - 

```
import { Alert } from '@chakra-ui/react'
...
<Alert.Title>...</Alert.Title>
```

and 

```
import { Alert } from '@chakra-ui/react'
...
const { Title } = Alert
... 
<Title>...</Title>
```

and also

```
import { AlertTitle } from '@chakra-ui/react'
... 
<AlertTitle>...</AlertTitle>
```

Are there any guidance as when to use of these over the other? Any pros or cons to any of these? Or are they all the exact same, just down to personal preference?

**Comment by segunadebayo** [comment]
All three imports resolve to the same component. Use `Alert.Title` when you want the parts grouped. Use `AlertTitle` when you want a flat import.

Destructuring `const { Title } = Alert` works, but the name is easy to collide with. Docs use the namespace form.
